### PR TITLE
V4 - Use new .NET CDN URLs and update to latest install scripts

### PR DESCRIPTION
**Pull request by heavymachinery**
**Description:**
- The .NET CDN URLs are changing. This PR updates `actions/setup-dotnet` to use the new URLs where needed.
- This PR also updates the action to use the the latest `install-dotnet` scripts which account for the new URLs. Note that these were pulled from the latest commit in https://github.com/dotnet/install-scripts as the new install scripts are not yet live on the https://dot.net website yet.

**Check list:**
- [ ] Mark if documentation changes are required.
- [ ] Mark if tests were added or updated to cover the changes.

### CODE REVIEW

**Comment by JamieMagee** [comment]
Related: https://github.com/dotnet/core/issues/9671

**Review by HarithaVattikuti — APPROVED** [review]

**Comment by Piedone** [comment]
Whoa, what a Christmas miracle that you fixed this before all of our .NET builds break, great job! Related: https://github.com/actions/runner-images/issues/11244.

**Comment by heavymachinery** [comment]
Thanks for the heads up @Piedone I've bubbled that issue up to the team 🙏

**Comment by Piedone** [comment]
Thank you!

**Comment by mungojam** [comment]
Thanks to all involved in getting this critical issue resolved so that most people won't even notice

File: dist/setup/index.js [hunk 1]
```diff
@@ -93821,7 +93827,8 @@ class DotnetVersionResolver {
     }
```
Comment (matejzvikl23) [inline dist/setup/index.js:93827]: GET https://github.com/login/oauth/authori

**Review by matejzvikl23 — COMMENTED** [review]
bj  by

**Pull request merged**